### ISSUE TITLE
Failed to create file with user.

### ISSUE THREAD

**Issue by ikeji** [itamae-kitchen/mitamae#71]
I got error when I user file resource with user attribute.

Here is my recipe:
```
file "/tmp/hoge" do
  user "ikeji"
  content "hoge"
end
```
"ikeji" is existing user.
I run this command in root.
```
# mitamae local hoge.rb
 INFO : Starting MItamae...
DEBUG : Loading recipe: hoge.rb
 INFO : Recipe: /tmp/hoge.rb
DEBUG :   file[/tmp/hoge]
DEBUG :     file[/tmp/hoge] action: create
DEBUG :       (in set_desired_attributes)
DEBUG :       (in set_current_attributes)
DEBUG :       (in pre_action)
DEBUG :       Executing `sudo -H -u ikeji -- /bin/sh -c cd\ \~ikeji\ \;\ diff\ -q\ /dev/null\ /tmp/1523180829.2237`...
DEBUG :         stderr | diff: /tmp/1523180829.2237: Permission denied
DEBUG :         exited with 2
ERROR :   file[/tmp/hoge] Failed.
```

possible work around is using owner attribute instead.
But that workaround doesn't work if mitamae is running as non root user.

**Comment by k0kubun** [comment]
This wasn't reproductive on my environment at all. I think it's not mitamae's bug but your environment's problem. Could you check the permission of `/tmp` and `/tmp/hoge`, or use `owner` if it doesn't work?

**Comment by ikeji** [comment]
Just FYI. This blog post seems similar problem.
http://jolantern.hatenablog.com/entry/2018/07/11/222244

**Comment by k0kubun** [comment]
Aha, now I probably seem to understand the problem. The behavior might not be intentional. I'll try to reproduce it again.

**Comment by k0kubun** [comment]
Now I can reproduce it and fix the bug. But at the same time, the article's way of using attributes (setting owner + group + user) is not quite intended (not the best way to use the `file` resource). 

Maybe you understand the problem (because you don't specify owner and group. Normally they should be specified instead of user though), but at least the article's author may not understand what `user` does.

**Comment by k0kubun** [comment]
@ikeji Sorry for your inconvenience by triaging this inappropriately :bow:

Now it's reproduced in https://github.com/itamae-kitchen/mitamae/commit/db6230f60c50529b9254ccda51da4d6a7b09fe59, fixed in https://github.com/itamae-kitchen/mitamae/commit/1f18586f807e872a655467dca39eeabdefb9f95d, and being released as v1.7.3 ([still in progress](https://travis-ci.org/itamae-kitchen/mitamae/builds/502894623)). Could you try that version?
